python_modules/dagster-graphql/dagster_graphql/implementation/fetch_assets.py

```diff
@@ -268,24 +268,38 @@ def get_asset_materializations(
     before_timestamp: Optional[float] = None,
     after_timestamp: Optional[float] = None,
     tags: Optional[Mapping[str, str]] = None,
+    storage_ids: Optional[Sequence[int]] = None,
 ) -> Sequence[EventLogEntry]:
     check.inst_param(asset_key, "asset_key", AssetKey)
     check.opt_int_param(limit, "limit")
     check.opt_float_param(before_timestamp, "before_timestamp")
     check.opt_mapping_param(tags, "tags", key_type=str, value_type=str)
 
     instance = graphene_info.context.instance
-    event_records = instance.get_event_records(
-        EventRecordsFilter(
-            event_type=DagsterEventType.ASSET_MATERIALIZATION,
-            asset_key=asset_key,
-            asset_partitions=partitions,
-            before_timestamp=before_timestamp,
-            after_timestamp=after_timestamp,
-            tags=tags,
-        ),
-        limit=limit,
+    records_filter = AssetRecordsFilter(
+        asset_key=asset_key,
+        asset_partitions=partitions,
+        before_timestamp=before_timestamp,
+        after_timestamp=after_timestamp,
+        tags=tags,
+        storage_ids=storage_ids,
     )
+    if limit is None:
+        event_records = []
+        cursor = None
+        while True:
+            event_records_result = instance.fetch_materializations(
+                records_filter=records_filter, cursor=cursor, limit=get_max_event_records_limit()
+            )
```

Comment:
specifically fetching this many materializations with the default limit (which is 10k I think?) still seems to time out at the postgres layer